tests/test_lora.py

```diff
@@ -277,3 +277,54 @@ def test_lora_qkv_linear_weights_merged_status(rank, enable_lora, expected_merge
     assert not layer.merged
     layer.merge()
     assert layer.merged == expected_merged
+
+
+@pytest.mark.skipif(torch.cuda.is_available() is False, reason="Quantization not supported on CPU. Skipping Test.")
+def test_lora_script_with_quantize(tmp_path, fake_checkpoint_dir, monkeypatch):
+    import finetune.lora as module
+
+    module.gradient_accumulation_iters = 1
+    module.save_interval = 2
+    module.eval_interval = 2
+    module.eval_iters = 2
+    module.max_iters = 6
+
+    data = [
+        {"input_ids": torch.tensor([0, 1, 2]), "labels": torch.tensor([1, 2, 3])},
+        {"input_ids": torch.tensor([1, 2, 3]), "labels": torch.tensor([2, 3, 4])},
+    ]
+    torch.save(data, tmp_path / "train.pt")
+    torch.save(data, tmp_path / "test.pt")
+
+    from lit_gpt.config import name_to_config
+
+    model_config = dict(block_size=128, n_layer=2, n_embd=8, n_head=4, padded_vocab_size=8)
+    monkeypatch.setitem(name_to_config, "tmp", model_config)
+
+    load_mock = Mock()
+    load_mock.return_value = load_mock
+    load_mock.__enter__ = Mock()
+    load_mock.__exit__ = Mock()
+    monkeypatch.setattr(module, "lazy_load", load_mock)
+
+    tokenizer_mock = Mock()
+    tokenizer_mock.return_value = tokenizer_mock
+    tokenizer_mock.encode = lambda *_, **kwargs: torch.tensor([3, 2, 1], **kwargs)
+    monkeypatch.setattr(module, "Tokenizer", tokenizer_mock)
+
+    stdout = StringIO()
+    with redirect_stdout(stdout):
+        module.setup(data_dir=tmp_path, checkpoint_dir=fake_checkpoint_dir, out_dir=tmp_path, precision="32-true", quantize="bnb.nf4")
+
+    assert {p.name for p in tmp_path.glob("*.pth")} == {
```

Comment:
This test is not valuable as it doesn't have any assertions specific to setting quantization. These are all already covered by the test above.

I wouldn't add it.